### ISSUE TITLE
Auto capitalize on a new line?

### ISSUE THREAD

**Issue by wpfan1000** [Julow/Unexpected-Keyboard#701]
Hi, if I have auto-capitalization turned on, and I type a new line, then the first letter is not capitalized.

I do see in Settings that it says to press shift on a new line.

I am just wondering why auto-capitalization is not enabled for new lines?

**Comment by Spike-from-NH** [comment]
I agree. No amount of white space (spaces/separators) should reset auto-capitalization.

(OP, it says to [auto-]press shift at the beginning of a new **sentence**—which is what we're talking about, new lines or not.)

**Comment by wpfan1000** [comment]
@Spike-from-NH Thanks for pointing that out - I edited my original post to be more accurate.

I am still wondering why auto-capitalization is not enabled for new lines? Is there a way to do this?

**Comment by Spike-from-NH** [comment]
There is currently no way to adjust the definition of auto-capitalization.  I believe we don't need user options but instead for the implementation to be made more useful, based on reactions such as yours.

**Comment by wpfan1000** [comment]
@Julow 
If I am not misunderstanding anything, please consider auto-capitalization for a new line.
Thanks

**Comment by Julow** [comment]
Auto-capitalization is implemented [here](https://github.com/Julow/Unexpected-Keyboard/blob/master/srcs/juloo.keyboard2/Autocapitalisation.java) if anyone has time to work on it.

**Comment by wpfan1000** [comment]
Unfortunately I am not coder so I cant help out with this.....

**Comment by Spike-from-NH** [comment]
In is_trigger_character(), line 155 seems to define space as the only trigger character. It should also include Enter and perhaps others. This is not a change I can make by imitation.

There is an additional oddity: Typing a second trigger character disables autocap, typing a third one re-enables it, and so on. We who learned on a typewriter to type two spaces at the end of a sentence are screwed.

**Comment by nahuhh** [comment]
It does work, just not properly.

- a new line does not detect as a new line
- type any letter, then delete it, and it will recognize as a new line 
fix: not sure. Seems like it should be possible to debug

related bug introduced by this feature
- autocapital enables the shift button
- the shift button interferes with `ctrl + a/c/x` etc
fix: pressing Ctrl should disable autoshift
